### PR TITLE
chore: update semantic convention version v1.27.0

### PR DESCRIPTION
**Pull request by lalitb**
## Changes

- Regular updates from the semantic convention v1.27.0:
    https://github.com/open-telemetry/semantic-conventions/releases/tag/v1.27.0
    https://github.com/open-telemetry/build-tools/releases/tag/v0.25.0

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-semantic-conventions/scripts/generate-consts-from-spec.sh [hunk 1]
```diff
@@ -35,7 +35,7 @@ docker run --rm \
 	-v "${CRATE_DIR}/scripts/templates:/templates" \
 	-v "${CRATE_DIR}/src:/output" \
 	otel/semconvgen:$SEMCOVGEN_VERSION \
-  --only span,event,scope \
```
Comment (lalitb) [inline opentelemetry-semantic-conventions/scripts/generate-consts-from-spec.sh:38]: `scope` type is removed from latest build-tool, and is unused - https://github.com/open-telemetry/build-tools/pull/309

**Comment by lalitb** [comment]
Moving to draft to fix the build error

File: opentelemetry-semantic-conventions/scripts/templates/semantic_attributes.rs.j2 [hunk 2]
```diff
@@ -14,7 +14,9 @@
 {%- set attribute = (attrs | selectattr('deprecated', 'none') | first) %}
 {%- set attribute = attribute if attribute else (attrs | first) %}
 {%- if conventions != 'attribute' %}
+{%- if not attribute.deprecated %}
```
Comment (lalitb) [inline opentelemetry-semantic-conventions/scripts/templates/semantic_attributes.rs.j2:17]: conditional check to ensure that the deprecated constants only defined with `#[deprecated]` annotation, and not exported.

**Comment by lalitb** [comment]
> Moving to draft to fix the build error

should be good now.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**